### ISSUE TITLE
Unable to authenticate with OIDC and Google OpenID connect

### ISSUE THREAD

**Issue by sandmule** [kiali/kiali#3410]
**General remarks**
We're trying to use google OIDC to authenticate into Kiali without RBAC. It is making the connection to google but is erroring in two places. 

**Describe the bug**
Inspecting the logs shows two errors: 

`[authentication.go:406] Token error: Unauthorized` - Where I believe the issue is coming from
`[authentication.go:688] Not handling OpenId code flow authentication: No nonce code present. Login window timed out.` - Don't believe this is the problem, as it's the flow handler.

I might be way off the mark here but as the first error appears to be coming from authentication.go:L406, I can see on line 399 it is attempting to get the namespaces. Is this with the credentials passed back? If so it won't work as OIDC isn't configured with the cluster due to using AWS managed Kubernetes.

**Versions used**
Kiali: 1.26.0
Istio: 1.7.3
Kubernetes flavour and version: EKS 1.18

**To Reproduce**
1. Configure OAUTH screen in google console
2. populate Kiali config
```
  auth:
    strategy: "openid"
    openid:
      # api_proxy: ''
      # api_proxy_ca_data: ''
      authentication_timeout: 600
      # authorization_endpoint: ''
      client_id: "XXXXXXX.apps.googleusercontent.com"
      disable_rbac: true
      # insecure_skip_verify_tls: false
      issuer_uri: 'https://accounts.google.com'
      scopes:
      - openid
      - email
      - profile
      username_claim: email
```
3. Create the secret inside the cluster
4. Add `X-Forwarded-Port: 443` to the gateway
5. Navigate to kiali endpoint and click on `Log in with OpenID`

**Expected behavior**
I expect to be able to log into kiali

Any help would be much appreciated, I've spent the day going through your codebase and exhausted all ideas.

**Comment by sandmule** [comment]
@israel-hdez I hope you don't mind the tag, but I saw you did a lot of the OIDC work and might be able to share some light on the problem

**Comment by israel-hdez** [comment]
I'll check. I think you are right that line 399 is the problem.
This may have slipped, as I think I had a fully integrated OIDC setup when I was working on this.

If that's the problem, I'll fix ASAP and push for a patch release of 1.26.

**Comment by sandmule** [comment]
Thank you very much for your fast response. More than happy to help with testing if required.

**Comment by israel-hdez** [comment]
@sandmule I have a fix in #3414 .

If you want to try the fix, you can try the image I uploaded in my quay.io repository: `quay.io/edgarhz/kiali:fix-3410`.
If you can confirm it works, I think I will be able to move it faster to a formal release.

**Comment by sandmule** [comment]
Yeah, just tested it and it working as intended 👍

**Comment by israel-hdez** [comment]
@sandmule The fix is released in 1.26.1.
If you use helm, charts are still pending. They should be ready soon.

But if you are using the operator (the recommended way) all is ready.
